taf/auth_repo.py

```diff
@@ -164,35 +181,26 @@ def _safely_get_json(self, commit, path):
             )
 
 
-class AuthenticationRepo(AuthRepoMixin, GitRepository):
+class AuthenticationRepo(GitRepository, AuthRepoMixin):
     def __init__(
-        self,
-        repo_path,
-        metadata_path="metadata",
-        targets_path="targets",
-        repo_urls=None,
-        additional_info=None,
-        default_branch="master",
+        self, repo_path, repo_urls=None, additional_info=None, default_branch="master"
     ):
         super().__init__(repo_path, repo_urls, additional_info, default_branch)
-        self.targets_path = targets_path
-        self.metadata_path = metadata_path
 
 
-class NamedAuthenticationRepo(AuthRepoMixin, NamedGitRepository):
+class NamedAuthenticationRepo(NamedGitRepository, AuthRepoMixin):
     def __init__(
         self,
         root_dir,
         repo_name,
-        metadata_path="metadata",
-        targets_path="targets",
         repo_urls=None,
         additional_info=None,
         default_branch="master",
     ):
-
         super().__init__(
-            root_dir, repo_name, repo_urls, additional_info, default_branch
+            root_dir=root_dir,
+            repo_name=repo_name,
```

Comment:
Looks good overall. Two main things that wee should consider:

- standardize parameter names - e.g. `repo_name`, `repository_name`, `repo`, `repository` (if class is called `Repository`, just `name` should be enough, but in case of multi-inheritance we could name it `repo_name` in subclass)
- creating a mechanism to create a `Repository` instance in previous revisions (contextmanager, pathing a tuf util function which reads metadata files)

Reply:
I believe that all repository classes contain the word `Repository`. I renamed `repo_name` to `name` and `repo_path` to `path`. We'll need to update all places in our code that extend `GitRepository`. Added a mechanism to instantiate TUF repository given an older commit